fix(image_projection_based_fusion): add run length decoding for segmentation_pointcloud_fusion

badai-nguyen opened the pull request:
## Description
- Since the segmentation mask will be encoded before publish as  PR: https://github.com/autowarefoundation/autoware.universe/pull/7905, so this PR is to add corresponding decoder for that.
- This PR need merge after #7905 merged.
- Related issue: https://github.com/autowarefoundation/autoware.universe/issues/5594

## Related links

**Parent Issue:**

- Link

<!-- ⬇️🟢
**Private Links:**

- [CompanyName internal link]()
⬆️🟢 -->

## How was this PR tested?

- Run `ros2 launch autoware_tensorrt_yolox yolox.launch.xml`  with `is_publish_color_mask: true`
- Run `ros2 launch autoware_image_projection_based_fusion segmentation_pointcloud_fusion.launch.xml` with `is_publish_debug_mask: true`
- Play rosbag (`input_bag` of an internal dataset) which contains `/sensing/camera/camera0/image_rect_color/compressed` and Pointcoud2 input topic for `/segmentation_pointcloud_fusion` node
- Confirmed the debug images by rqt that have the same segmentation result as below (Note: the colors are different because of different color map was used)  
![image](https://github.com/user-attachments/assets/79064d05-5f82-48a2-8e89-7db2a2e236e0)


## Notes for reviewers

None.

## Interface changes

None.

<!-- ⬇️🔴

### Topic changes

#### Additions and removals

| Change type   | Topic Type      | Topic Name    | Message Type        | Description       |
|:--------------|:----------------|:--------------|:--------------------|:------------------|
| Added/Removed | Pub/Sub/Srv/Cli | `/topic_name` | `std_msgs/String`   | Topic description |

#### Modifications

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Pub/Sub/Srv/Cli | `/old_topic_name` | `sensor_msgs/Image` | Topic description |
| New     | Pub/Sub/Srv/Cli | `/new_topic_name` | `sensor_msgs/Image` | Topic description |

### ROS Parameter Changes

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added/Removed | `param_name`   | `double` | `1.0`         | Param description |

#### Modifications

| Version | Parameter Name   | Type     | Default Value | Description       |
|:--------|:-----------------|:---------|:--------------|:------------------|
| Old     | `old_param_name` | `double` | `1.0`         | Param description |
| New     | `new_param_name` | `double` | `1.0`         | Param description |

🔴⬆️ -->

## Effects on system behavior

None.

kminoda commented on perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp:
```diff
@@ -57,19 +60,23 @@ void SegmentPointCloudFusionNode::fuseOnSingleImage(
+    debug_mask_msg->header = input_mask.header;
+    pub_debug_mask_ptr_.publish(debug_mask_msg);
   }
+  Eigen::Matrix4d projection;
+  projection << camera_info.p.at(0), camera_info.p.at(1), camera_info.p.at(2), camera_info.p.at(3),
+    camera_info.p.at(4), camera_info.p.at(5), camera_info.p.at(6), camera_info.p.at(7),
+    camera_info.p.at(8), camera_info.p.at(9), camera_info.p.at(10), camera_info.p.at(11), 0.0, 0.0,
+    0.0, 1.0;
```
Is this used somewhere?

badai-nguyen replied:
@kminoda thank you for you comment. I my mistake when rebased branch. I fixed t https://github.com/autowarefoundation/autoware.universe/pull/7909/commits/b8abe57a82d8655e33bcb24e7ba2e9e2a69ef7fc

kminoda commented on perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp:
```diff
@@ -16,6 +16,7 @@
 
 #include "autoware/image_projection_based_fusion/utils/geometry.hpp"
 #include "autoware/image_projection_based_fusion/utils/utils.hpp"
+#include "perception_utils/run_length_encoder.hpp"
```
```suggestion
#include <perception_utils/run_length_encoder.hpp>
```

kminoda reviewed (commented):
I added one small comment :pray: 

Also, please update `How was this PR tested?` section.

YoshiRi reviewed (approved):
LGTM

badai-nguyen commented:
> I added one small comment 🙏
> 
> Also, please update `How was this PR tested?` section.

@kminoda thanks for your comment.
I updated `How was PR tested`. Could you check again. 
By the way, I also open PR for unit test to confirm the output after decoder the same with the input image here :https://github.com/autowarefoundation/autoware.universe/pull/8414

badai-nguyen commented:
@kminoda Do you have other comments? Otherwise, I will merge this PR.
